craft_providers/bases/buildd.py

```diff
@@ -0,0 +1,522 @@
+        the environment.
+
+        If timeout is specified, abort operation if time has been exceeded.
+
+        Guarantees provided by this setup:
+
+            - configured hostname
+
+            - networking available (IP & DNS resolution)
+
+            - apt cache up-to-date
+
+            - snapd configured and ready
+
+            - system services are started and ready
+
+        :param executor: Executor for target container.
+        :param retry_wait: Duration to sleep() between status checks (if
+            required).
+        :param timeout: Timeout in seconds.
+
+        :raises BaseConfigurationError: on other unexpected error.
+        """
+        if timeout is not None:
+            deadline: Optional[float] = time.time() + timeout
+        else:
+            deadline = None
+
+        self._setup_environment(executor=executor, deadline=deadline)
+        self._setup_wait_for_system_ready(
+            executor=executor, deadline=deadline, retry_wait=retry_wait
+        )
+        self._setup_hostname(executor=executor, deadline=deadline)
+        self._setup_resolved(executor=executor, deadline=deadline)
+        self._setup_networkd(executor=executor, deadline=deadline)
+        self._setup_wait_for_network(
+            executor=executor, deadline=deadline, retry_wait=retry_wait
+        )
+        self._setup_apt(executor=executor, deadline=deadline)
+        self._setup_snapd(executor=executor, deadline=deadline)
+
+    def _setup_apt(self, *, executor: Executor, deadline: Optional[float]) -> None:
+        """Configure apt & update cache.
+
+        :param executor: Executor for target container.
+        """
+        executor.create_file(
+            destination=pathlib.Path("/etc/apt/apt.conf.d/00no-recommends"),
+            content=io.BytesIO('Apt::Install-Recommends "false";\n'.encode()),
+            file_mode="0644",
+        )
+
+        try:
+            executor.execute_run(
+                ["apt-get", "update"],
+                capture_output=True,
+                check=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to update apt cache.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        try:
+            executor.execute_run(
+                ["apt-get", "install", "-y", "apt-utils"],
+                capture_output=True,
+                check=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to install apt-utils.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        _check_deadline(deadline)
```

Comment:
Is this last check necessary? It already checked in the previous `try` block.

Reply:
Fixed!